common/src/main/java/com/revenuecat/purchases/common/networking/ETagManager.kt

```diff
@@ -0,0 +1,99 @@
+package com.revenuecat.purchases.common.networking
+
+import android.content.Context
+import android.content.SharedPreferences
+import org.json.JSONObject
+import java.net.HttpURLConnection
+
+private const val SERIALIZATION_NAME_ETAG = "eTag"
+private const val SERIALIZATION_NAME_HTTPRESULT = "httpResult"
+
+internal const val ETAG_HEADER_NAME = "X-RevenueCat-ETag"
+
+data class HTTPResultWithETag(
+    val eTag: String,
+    val httpResult: HTTPResult
+) {
+    fun serialize(): String {
+        return JSONObject().apply {
+            put(SERIALIZATION_NAME_ETAG, eTag)
+            put(SERIALIZATION_NAME_HTTPRESULT, httpResult.serialize())
+        }.toString()
+    }
+
+    companion object {
+        fun deserialize(serialized: String): HTTPResultWithETag {
+            val jsonObject = JSONObject(serialized)
+            val eTag = jsonObject.getString(SERIALIZATION_NAME_ETAG)
+            val serializedHTTPResult = jsonObject.getString(SERIALIZATION_NAME_HTTPRESULT)
+            return HTTPResultWithETag(eTag, HTTPResult.deserialize(serializedHTTPResult))
+        }
+    }
+}
+
+class ETagManager(
+    private val prefs: SharedPreferences
+) {
+
+    internal fun getETagHeader(
+        path: String,
+        refreshETag: Boolean = false
+    ): Map<String, String> {
+        val eTagHeader = ETAG_HEADER_NAME to if (refreshETag) "" else getETag(path)
+        return mapOf(eTagHeader)
+    }
+
+    internal fun shouldUseCachedVersion(responseCode: Int) = responseCode == RCHTTPStatusCodes.NOT_MODIFIED
+
+    internal fun getStoredResult(path: String): HTTPResult? {
+        val storedResult = getStoredResultSavedInSharedPreferences(path)
+        return storedResult?.httpResult
+    }
+
+    internal fun storeBackendResultIfNoError(
+        path: String,
+        resultFromBackend: HTTPResult,
+        eTagInResponse: String
+    ) {
+        val responseCode = resultFromBackend.responseCode
+        if (responseCode != RCHTTPStatusCodes.NOT_MODIFIED && responseCode < RCHTTPStatusCodes.ERROR) {
```

Comment:
out of curiosity, why the `RC` prefix? We don't usually add them in Kotlin, right?

Reply:
Becuase since it's public it will be exposed to everyone as HTTPStatusCodes, which I believe is a very common name.

Reply:
Yeah, due to that thing we found out about maven dependencies not respecting `implementation` vs `api`. Definitely something we need to figure out rather soon.